File: Financial Calculator/update_budget.py
def user_input(type):
    return(float(input(f"How much is your {type}?: ")))

print("This is going to calculate your budget for the month")

income = user_input("income")
rent = user_input("rent")
utilities = user_input("utilities")
groceries = user_input("groceries")
transportation = user_input("transportation")

expenses = rent + utilities + groceries + transportation
savings = float(income) *.2
total = income - expenses - savings

def percent(type, amount):
    per = int(amount / income *100)
    return f"Your {type} is {per}% of your income"

print("Your income is", income)
print("Your expenses are", expenses)
print("Your savings are", savings)
print("Your total left to spend is", total)

print(percent("rent", rent))
print(percent("utilities", utilities))
print(percent("groceries", groceries))
print(percent("transporation", transportation))
print(percent("expenses", expenses))

File: Financial Calculator/update_budget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, expenses, savings, total;

float input(char type[], float var){
    printf("How much is your %s: \n", type);
    scanf("%f", &var);
    return var;
}

void percent(char type[], int amount){
    int per = amount/income *100;
    printf("Your %s is %d%% of your income\n" , type, per);
}


int main(void){
    printf("This is going to calculate your budget for the month \n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);

    expenses = rent + utilities + groceries + transportation;
    savings = income * .2;
    total = income - expenses - savings;

    printf("Your income is: $%.2f\n", income); //f is float .2 is how many decimal points
    printf("Your expenses are: $%.2f\n", expenses);
    printf("Your savings are: $%.2f\n", savings);
    printf("Your total left to spend is $%.2f\n", total);
    
    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("savings", savings);
    percent("expenses", expenses);

    return 0;
    }